cypress-io/cypress
# feat: WIP no merge - add ipc to allow outside processes to communicate with cypress (WIP)

- Closes N/A

### Additional details


Adds an IPC that any other process can connect to and request the endpoint for the GraphQL API. This will let our VS Code extension either

1. Get the current Cypress instance's GraphQL endpoint
2. If it fails, we can launch Cypress from the extension (this has not been implemented yet)

Matching extension PR: https://github.com/cypress-io/cypress-extensions/pull/14

### Steps to test


1. Get this branch
2. Install the latest VS Code extension (grab `cypress.vsix` from [this PR]( https://github.com/cypress-io/cypress-extensions/pull/14)). You can load it, see image below.
3. Open Cypress in a project (say packages/app)
4. Pick a spec and run "Cypress: Open in Cypress" from the command palette

If Cypress is NOT open, VS Code will give an error informing the user. In the future, we will spawn Cypress for the user automatically.

![image](https://github.com/cypress-io/cypress/assets/19196536/18b9059e-e9c4-4763-9e67-f040a4cb8e7d)

### How has the user experience changed?


Does not require extension to hard-code the GraphQL endpoint. This will make the extension usable for any production project.

### PR Tasks
